### PR TITLE
v4 and tRPC 10 support

### PR DESCRIPTION
**Pull request by wobsoriano**
https://trpc-nuxt-beta.vercel.app/

### CODE REVIEW

**Comment by Q16solver** [comment]
Hello, I have a question about how the useAsyncData under the hood works. Since it's always async, that means that after querying/mutating, the data object that gets returned could be null and doesn't immediately get the result after awaiting? If I have an onClick that calls this, will it be able to actually get the data back instead of null (also doesn't this mean I have to always to null if checks)

image

**Comment by Q16solver** [comment]
Also a different issue, this might be to do with upstream trpc type inference but I'm not sure... The issue is if we decide to do some ternary condition in the return value from the backend, the type is inferred as Pick<Object, never>, but if I don't do that, it's fine, and infers Object type properly

image
image

**Review by juls0730 — APPROVED** [review]
Outstanding work, you have the go ahead from me!

**Comment by Q16solver** [comment]
The init trpc also seems to be run on the client as well, is there any easy way to have it only run on the server? I definitely would've thought that nuxt 3 runs everything in the `server` folder only on the server, I guess I was wrong?

image

File: docs/content/1.get-started/2.usage.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+---
+title: Usage
+description: tRPC-Nuxt provides first class integration with tRPC.
+---
+
+# Usage
+
+## Recommended file structure
+
+Recommended but not enforced file structure. This is what you get when starting from [the examples](../main/example-apps.md).
```
Comment (CodyBontecou) [inline docs/content/1.get-started/2.usage.md:10]: Looks like this is following an older pattern. This is now 404ing.

**Pull request merged**